registry/registry_test.go

```diff
@@ -0,0 +1,54 @@
+package registry
+
+import (
+	"testing"
+
+	"github.com/OrlandoRomo/academy-go-q32021/infrastructure/api"
+	"github.com/OrlandoRomo/academy-go-q32021/interface/controller"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testApiKey = "cc2464e4a1mshb5ceeca91e5a6adp1fa80bjsn4b48e2408b87"
+)
+
+func TestNewRegistry(t *testing.T) {
+	testcases := []struct {
+		name        string
+		message     string
+		urbanClient *api.UrbanDictionary
+	}{
+		{
+			name:        "success - urban client set up",
+			message:     "urban client in registry set up",
+			urbanClient: api.NewUrbanDictionary(testApiKey),
+		},
+		{
+			name:        "failure - urban client nil",
+			message:     "registry app with a nil urban client",
+			urbanClient: nil,
+		},
+	}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.urbanClient == nil {
+				assert.Nil(t, test.urbanClient, test.message)
+			}
+			if test.urbanClient != nil {
+				assert.NotNil(t, test.urbanClient, test.message)
+			}
+		})
+	}
+}
+
+func TestNewAppController(t *testing.T) {
+	appController := NewRegistry(nil).NewAppController()
```

Comment:
I find it a bit odd to use the `for` template for some tests, and then this individual `t.Run(...)` format on different tests, please be consistent, I would recommend using always the `for` approach